src/mjolnir/bssbuilder.cc

```diff
@@ -95,23 +95,6 @@ struct BSSConnection {
   }
 };
 
-template <typename T> struct Finally {
-  T t;
-  explicit Finally(T t) : t(t){};
-  Finally() = delete;
-  Finally(Finally&& f) = default;
-  Finally(const Finally&) = delete;
-  Finally& operator=(const Finally&) = delete;
-  Finally& operator=(Finally&&) = delete;
-  ~Finally() {
-    t();
-  };
-};
-
-template <typename T> Finally<T> make_finally(T t) {
-  return Finally<T>{t};
-};
```

Comment:
this RAII based approach to implementing something like javas finally was exactly the same one i had previously implemented for the scoped_timer. since @xlqian 's version here is more generic, i decided to move it to midgard and then use it to replace the scoped_timer. great minds think alike!